nanopub/nanopub.py

```diff
@@ -200,114 +207,111 @@ class NanopubClient:
+        params = dict()
+        params['type'] = type
+        params['searchterm'] = searchterm
 
-        searchparams['type'] = thing_type
-        searchparams['searchterm'] = searchterm
+        return self._search(endpoint='find_things',
+                            params=params,
+                            max_num_results=max_num_results, )
 
-        return self._search(searchparams=searchparams,
-                            max_num_results=max_num_results, apiurl=apiurl)
+    def _query_grlc(self, params, endpoint):
+        """Query the nanopub server grlc endpoint.
 
-    @staticmethod
-    def _search(searchparams=None, max_num_results=None, apiurl=None):
+        Query a nanopub grlc server endpoint (for example: find_text). Try several of the nanopub
+        garlic servers.
         """
-        General nanopub server search method. User should use e.g. search_text() or search_pattern() instead.
-        """
-
-        if apiurl is None:
-            raise ValueError('kwarg "apiurl" must be specified. Consider using search_text() function instead.')
-
-        if max_num_results is None:
-            raise ValueError('kwarg "max_num_results" must be specified. Consider using search_text() function instead.')
-
-        if searchparams is None:
-            raise ValueError('kwarg "searchparams" must be specified. Consider using search_text() function instead.')
-
-
-        # Query the nanopub server for the specified text
         headers = {"Accept": "application/json"}
-        r = requests.get(apiurl, params=searchparams, headers=headers)
+        r = None
+        for grlc_url in self.grlc_urls:
+            url = grlc_url + endpoint
+            r = requests.get(url, params=params, headers=headers)
+            if r.status_code == 502:  # Server is likely down
+                warnings.warn(f'Could not get response from {grlc_url}, trying other servers')
+            else:
+                r.raise_for_status()  # For non-502 errors we don't want to try other servers
+                return r
 
-        if r.ok:
+        raise requests.HTTPError(f'Could not get response from any of the nanopub grlc '
+                                 f'endpoints, last response: {r.status_code}:{r.reason}')
 
-            # Make sure that results are provided
-            try:
-                results_json = r.json()
-            except:
-                # If the returned message can't be serialized as JSON (such as due to virtuoso error) then there are no results
-                print('Error: Could not serialize response as JSON:\n', r.content)
-                return []
+    def _search(self, endpoint, params, max_num_results):
+        """
+        General nanopub server search method. User should use e.g. search_text() or search_pattern() instead.
 
-            results_list = results_json['results']['bindings']
-            nanopubs = []
+        Raises:
```

Comment:
Maybe include the arguments in the docstring, and what form they should take (I guess only ```params``` is likely to be unclear)

Reply:
Check!